camel/storages/dict_storage/in_memory.py

```diff
@@ -0,0 +1,38 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+from copy import deepcopy
+from typing import Any, Dict, List
+
+from camel.storages.dict_storage.base import BaseDictStorage
+
+
+class InMemoryDictStorage(BaseDictStorage):
```

Comment:
What is it a `DictStorage`? It seems to be just `InMemoryStorage` because it is not a persistent storage.

Reply:
Different from `VectorDBStorage`, `SQLDBStorage`, or `KnowledgeGraphStorage`, `DictStorage` just stores the data structure "dictionary". It can have many backends like `Json` or `MangoDB`. `InMemory` is just a type of backend. BTW, `VectorDBStorage` can also be in-memory.

Reply:
A better name can be `KeyValueStorage`, but the current interefaces are ugly and are not actually "key-value". We may refactor it later with "key-value" paradigm. For now we just leave it, since this PR is already too large.